include/deal.II/grid/grid_tools.h

```diff
@@ -461,6 +461,35 @@ namespace GridTools
                       const double max_ratio = 1.6180339887,
                       const unsigned int max_iterations = 5);
 
+  /**
+   * Analyze the boundary cells of a mesh, and if one cell is found at
+   * a corner position (with dim adjacent faces on the boundary), and its
+   * dim-dimensional angle fraction exceeds @p limit_angle_fraction,
+   * refine globally once, and replace the children of such cell
+   * with children where the corner is no longer offending the given angle
+   * fraction.
```

Comment:
could you please add a two image which illustrate this in 2d? It was not obvious to me what

> replace the children of such cell with children where the corner is no longer offending the given angle fraction.

means.